pkgs/development/tools/continuous-integration/github-runner/default.nix

```
@@ -276,6 +273,34 @@ stdenv.mkDerivation rec {
     wrap config.sh --prefix PATH : ${lib.makeBinPath [ glibc.bin ]}
   '';
 
+  # Script to create deps.nix file for dotnet dependencies. Run it with
+  # $(nix-build -A github-runner.passthru.createDepsFile)/bin/create-deps-file
+  # and copy result file into github-runner folder.
+  # Inspired by passthru.fetch-deps in pkgs/build-support/build-dotnet-module/default.nix
+  passthru.createDepsFile = writeShellApplication {
+    name = "create-deps-file";
+    runtimeInputs = [ dotnetSdk nuget-to-nix ];
+    text = ''
+      printf "\n* Setup workdir\n"
+      workdir="$(mktemp -d /tmp/github-runner.XXX)"
+      cp -rT "${src}" "$workdir"
+      chmod -R +w "$workdir"
+      trap 'rm -rf "$workdir"' EXIT
+
+      pushd "$workdir"
+
+      mkdir nuget_pkgs
+
+      printf "\n* Restore ${pname} dotnet project\n"
+      dotnet restore src/ActionsRunner.sln --packages nuget_pkgs --no-cache --force --runtime linux-x64
+
+      printf "\n* Make deps.nix file\n"
+      deps_file="/tmp/${pname}-deps.nix"
+      nuget-to-nix nuget_pkgs > "$deps_file"
+      printf "\n* Dependency file writen to %s" "$deps_file"
+     '';
```

Minor indention nit:

```suggestion
    '';
```